### PR TITLE
document the format of DSA signature in EVP_SIGNATURE-DSA

### PR DESCRIPTION
**Pull request by vladak**
This trivial change documents the DSA signature format produced with `EVP_PKEY_sign()`. I found it handy since I needed to do subsequent processing of the signature.

### CODE REVIEW

**Comment by slontis** [comment]
Wouldnt it be better to cover other keys also?

Maybe by putting in a section in EVP_PKEY_sign() 
like this?
````
=head1 CONFORMING TO

RFC 3279 Section 2.2 Signature Algorithms
````

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
IMO it is fine here.

**Comment by slontis** [comment]
@t8m why is being inconsistent across different similar files fine?

**Comment by t8m** [comment]
This is just adding more information to documentation. I do not see the inconsistency or any harm caused. BTW the signature format for for example RSA is more complex so it would not be correct to add this to EVP_PKEY_sign().

**Comment by paulidale** [comment]
I agree with @slontis, we should be consistent.  The hodgepodge nature of our documentations is a negative that's been mentioned multiple times in the past.

**Comment by levitte** [comment]
We can't really say much about conformance with certain RFCs in the EVP_PKEY_sign documentation, because that excludes all signature algorithms that aren't part of the RFC in question.

And really, conformance belongs with the provider implementation and it's documentation rather than with the general EVP functions.

I agree that there's a hodge podge, and would argue that it's partly because implementation specific documentation has found it's way to higher level EVP documentation, *where it doesn't really belong*

**Comment by t8m** [comment]
> We can't really say much about conformance with certain RFCs in the EVP_PKEY_sign documentation, because that excludes all signature algorithms that aren't part of the RFC in question.
> 
> And really, conformance belongs with the provider implementation and it's documentation rather than with the general EVP functions.
> 
> I agree that there's a hodge podge, and would argue that it's partly because implementation specific documentation has found it's way to higher level EVP documentation, _where it doesn't really belong_

This is adding the documentation to EVP_SIGNATURE-DSA and that is IMO the right place for it. Random providers SHOULD NOT produce a different format for signatures for the same algorithm.

**Comment by t8m** [comment]
So ping for @openssl/committers second review

**Review by t-j-h — APPROVED** [review]

**Comment by t-j-h** [comment]
Okay for doc backfit to whichever branches you see as relevant ...

**Comment by levitte** [comment]
> This is adding the documentation to EVP_SIGNATURE-DSA and that is IMO the right place for it. Random providers SHOULD NOT produce a different format for signatures for the same algorithm.

Good point.  But then, you can argue my point by changing "implementation specific" to "algorithm specific", and you still get to the same conclusion regarding where conformance specification should end up.

**Comment by t8m** [comment]
> > This is adding the documentation to EVP_SIGNATURE-DSA and that is IMO the right place for it. Random providers SHOULD NOT produce a different format for signatures for the same algorithm.
> 
> Good point. But then, you can argue my point by changing "implementation specific" to "algorithm specific", and you still get to the same conclusion regarding where conformance specification should end up.

Yes, and this PR is exactly about that - it is adding the spec into EVP_SIGNATURE-DSA which is "algorithm specific" documentation.

**Comment by levitte** [comment]
Yup. I'm not protesting this PR, but some of the earlier discussion

**Comment by t8m** [comment]
Merged to all the active branches. Thank you for your contribution.

**Pull request closed without merging**
